tests/unit/s2n_tls13_cert_verify_test.c

```diff
@@ -63,38 +63,40 @@ int run_tests(const struct s2n_tls13_cert_verify_test *test_case, s2n_mode verif
 
     EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, cert_chain));
 
-    /* Successfully send and receive certificate verify */
-    {
-        /* Derive private/public keys and set connection variables */
-        struct s2n_stuffer certificate_in = { 0 }, certificate_out = { 0 };
-        struct s2n_blob b = { 0 };
-        s2n_pkey_type pkey_type = { 0 };
+    /* Initialize a certificate */
+    DEFER_CLEANUP(struct s2n_stuffer certificate_in = { 0 }, s2n_stuffer_free);
+    DEFER_CLEANUP(struct s2n_stuffer certificate_out = { 0 }, s2n_stuffer_free);
+    struct s2n_blob b = { 0 };
+    s2n_pkey_type pkey_type = { 0 };
 
-        struct s2n_connection *verifying_conn = NULL, *sending_conn = NULL;
-        EXPECT_NOT_NULL(verifying_conn = s2n_connection_new(verifier_mode));
-        verifying_conn->actual_protocol_version = S2N_TLS13;
-        EXPECT_NOT_NULL(sending_conn = s2n_connection_new(verifier_mode == S2N_CLIENT ? S2N_SERVER : S2N_CLIENT));
+    EXPECT_SUCCESS(s2n_blob_init(&b, (uint8_t *) cert_chain_pem, strlen(cert_chain_pem) + 1));
+    EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_in, S2N_MAX_TEST_PEM_SIZE));
+    EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_out, S2N_MAX_TEST_PEM_SIZE));
+    EXPECT_SUCCESS(s2n_stuffer_write(&certificate_in, &b));
+    EXPECT_SUCCESS(s2n_stuffer_certificate_from_pem(&certificate_in, &certificate_out));
 
-        EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_in, S2N_MAX_TEST_PEM_SIZE));
-        EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_out, S2N_MAX_TEST_PEM_SIZE));
+    uint32_t available_size = s2n_stuffer_data_available(&certificate_out);
+    EXPECT_SUCCESS(s2n_blob_init(&b, s2n_stuffer_raw_read(&certificate_out, available_size), available_size));
 
-        EXPECT_SUCCESS(s2n_connection_set_config(sending_conn, config));
+    /* Successfully send and receive certificate verify */
+    {
+        DEFER_CLEANUP(struct s2n_connection *sending_conn = s2n_connection_new(verifier_mode == S2N_CLIENT ? S2N_SERVER : S2N_CLIENT), s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(sending_conn);
```

Comment:
https://github.com/aws/s2n-tls/blob/main/tls/s2n_connection.h#L48, and watch for unreasonably long lines
```suggestion
    {
        DEFER_CLEANUP(struct s2n_connection *sending_conn = s2n_connection_new(S2N_PEER_MODE(verifier_mode),
                s2n_connection_ptr_free);
        EXPECT_NOT_NULL(sending_conn);
```